metadata-ingestion/src/datahub/utilities/sqlglot_lineage.py

```diff
@@ -774,17 +795,43 @@ def _extract_select_from_update(
             # they'll get caught later.
             new_expressions.append(expr)
 
-    return sqlglot.exp.Select(
+    # Special translation for the `from` clause.
+    extra_args = {}
+    original_from = statement.args.get("from")
+    if original_from and isinstance(original_from.this, sqlglot.exp.Table):
+        # Move joins, laterals, and pivots from the Update->From->Table->field
+        # to the top-level Select->field.
+
+        for k in _UPDATE_FROM_TABLE_ARGS_TO_MOVE:
+            if k in original_from.this.args:
+                # Mutate the from table clause in-place.
+                extra_args[k] = original_from.this.args.pop(k)
+
+    select_statement = sqlglot.exp.Select(
         **{
             **{
                 k: v
                 for k, v in statement.args.items()
                 if k not in _UPDATE_ARGS_NOT_SUPPORTED_BY_SELECT
             },
+            **extra_args,
             "expressions": new_expressions,
         }
     )
 
+    # Update statements always implicitly have the updated table in context.
+    # TODO: Retain table name alias.
+    if select_statement.args.get("from"):
+        # select_statement = sqlglot.parse_one(select_statement.sql(dialect=dialect))
```

Comment:
Remove this?

Reply:
will do in a follow-up

Reply:
done in here https://github.com/datahub-project/datahub/pull/9119/commits/87d0b06232843726973d744b40539ce0ec5d4151